<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Functions to Create Publication-Quality Plots • BoutrosLab.plotting.general</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="deps/headroom-0.11.0/headroom.min.js"></script><script src="deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="deps/search-1.0.0/fuse.min.js"></script><script src="deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Functions to Create Publication-Quality Plots">
<meta name="description" content="Contains several plotting functions such as barplots, scatterplots, heatmaps, as well as functions to combine plots and assist in the creation of these plots. These functions will give users great ease of use and customization options in broad use for biomedical applications, as well as general purpose plotting. Each of the functions also provides valid default settings to make plotting data more efficient and producing high quality plots with standard colour schemes simpler. All functions within this package are capable of producing plots that are of the quality to be presented in scientific publications and journals. P'ng et al.; BPG: Seamless, automated and interactive visualization of scientific data; BMC Bioinformatics 2019 &lt;doi:10.1186/s12859-019-2610-2&gt;.">
<meta property="og:description" content="Contains several plotting functions such as barplots, scatterplots, heatmaps, as well as functions to combine plots and assist in the creation of these plots. These functions will give users great ease of use and customization options in broad use for biomedical applications, as well as general purpose plotting. Each of the functions also provides valid default settings to make plotting data more efficient and producing high quality plots with standard colour schemes simpler. All functions within this package are capable of producing plots that are of the quality to be presented in scientific publications and journals. P'ng et al.; BPG: Seamless, automated and interactive visualization of scientific data; BMC Bioinformatics 2019 &lt;doi:10.1186/s12859-019-2610-2&gt;.">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="index.html">BoutrosLab.plotting.general</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">7.1.3</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/uclahs-cds/package-BoutrosLab-plotting-general/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header"><h1 id="boutroslabplottinggeneral">BoutrosLab.plotting.general<a class="anchor" aria-label="anchor" href="#boutroslabplottinggeneral"></a>
</h1></div>
<ol style="list-style-type: decimal">
<li><a href="#description">Description</a></li>
<li><a href="#included-plots">Included plots</a></li>
<li><a href="#installation">Installation</a></li>
<li><a href="#function-reference">Function reference</a></li>
<li><a href="#resources">Resources</a></li>
<li><a href="#getting-help">Getting help</a></li>
<li><a href="#contributors">Contributors</a></li>
<li><a href="#citation-information">Citation information</a></li>
<li><a href="#license">License</a></li>
</ol>
<div class="section level2">
<h2 id="description">Description<a class="anchor" aria-label="anchor" href="#description"></a>
</h2>
<p>Contains several plotting functions such as barplots, scatterplots, heatmaps, as well as functions to combine plots and assist in the creation of these plots. These functions will give users great ease of use and customization options in broad use for biomedical applications, as well as general purpose plotting. Each of the functions also provides valid default settings to make plotting data more efficient and producing high quality plots with standard colour schemes simpler. All functions within this package are capable of producing plots that are of the quality to be presented in scientific publications and journals. P’ng et al.; <a href="doi:10.1186/s12859-019-2610-2">BPG: Seamless, automated and interactive visualization of scientific data</a>; BMC Bioinformatics 2019.</p>
</div>
<div class="section level2">
<h2 id="included-plots">Included plots<a class="anchor" aria-label="anchor" href="#included-plots"></a>
</h2>
<p><img src="https://media.springernature.com/full/springer-static/image/art%3A10.1186%2Fs12859-019-2610-2/MediaObjects/12859_2019_2610_Fig1_HTML.png?as=webp"></p>
<p>Available chart-types. The basic chart-types available in BPG: <strong>a</strong> density plot, <strong>b</strong> boxplot, <strong>c</strong> violin plot, <strong>d</strong> segplot, <strong>e</strong> strip plot, <strong>f</strong> barplot, <strong>g</strong> scatterplot, <strong>h</strong> histogram, <strong>i</strong> qqplot fit, <strong>j</strong> qqplot comparison, <strong>k</strong> Manhattan plot, <strong>l</strong> polygon plot, <strong>m</strong> heatmap, <strong>n</strong> dotmap and <strong>o</strong> hexbinplot.</p>
<p><img src="https://media.springernature.com/full/springer-static/image/art%3A10.1186%2Fs12859-019-2610-2/MediaObjects/12859_2019_2610_Fig2_HTML.png?as=webp"></p>
<p>Multiplot example. BPG’s create.multiplot function is able to combine multiple chart-types together into a single figure as shown above.</p>
<blockquote>
<p><strong>Note</strong>: Both <code>create.multipanelplot</code> and <code>create.multiplot</code> are currently supported, however <code>create.multipanelplot</code> is preferable to <code>create.multiplot</code>. In the medium-term <code>create.multiplot</code> will be deprecated.</p>
</blockquote>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>To install the latest public release of BoutrosLab.plotting.general from CRAN:</p>
<pre><code><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BoutrosLab.plotting.general"</span><span class="op">)</span></span></code></pre>
<p>Or to install the latest development version from GitHub:</p>
<pre><code><span><span class="co"># install.packages("devtools")</span></span>
<span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"uclahs-cds/package-BoutrosLab-plotting-general"</span><span class="op">)</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="function-reference">Function reference<a class="anchor" aria-label="anchor" href="#function-reference"></a>
</h2>
<p>For further details on usage, parameters or to see an example of a specific function, see function links below.</p>
<table class="table">
<colgroup>
<col width="55%">
<col width="45%">
</colgroup>
<thead><tr class="header">
<th><strong>Functions</strong></th>
<th><strong>Description</strong></th>
</tr></thead>
<tbody>
<tr class="odd">
<td><a href="https://uclahs-cds.github.io/package-BoutrosLab-plotting-general/reference/auto.axis.html" class="external-link"><code>auto.axis()</code></a></td>
<td>Create ideal labels and values for a given dataset (detects log scales)</td>
</tr>
<tr class="even">
<td><a href="https://uclahs-cds.github.io/package-BoutrosLab-plotting-general/reference/colour.gradient.html" class="external-link"><code>colour.gradient()</code></a></td>
<td>Creates a colour gradient</td>
</tr>
<tr class="odd">
<td><a href="https://uclahs-cds.github.io/package-BoutrosLab-plotting-general/reference/covariates.grob.html" class="external-link"><code>covariates.grob()</code></a></td>
<td>Create one or more covariate bars</td>
</tr>
<tr class="even">
<td><a href="https://uclahs-cds.github.io/package-BoutrosLab-plotting-general/reference/create.barplot.html" class="external-link"><code>create.barplot()</code></a></td>
<td>Make a barplot</td>
</tr>
<tr class="odd">
<td><a href="https://uclahs-cds.github.io/package-BoutrosLab-plotting-general/reference/create.boxplot.html" class="external-link"><code>create.boxplot()</code></a></td>
<td>Make a boxplot</td>
</tr>
<tr class="even">
<td><a href="https://uclahs-cds.github.io/package-BoutrosLab-plotting-general/reference/create.colourkey.html" class="external-link"><code>create.colourkey()</code></a></td>
<td>Create Colourkey</td>
</tr>
<tr class="odd">
<td><a href="https://uclahs-cds.github.io/package-BoutrosLab-plotting-general/reference/create.dendrogram.html" class="external-link"><code>create.dendrogram()</code></a></td>
<td>Generate a dendrogram</td>
</tr>
<tr class="even">
<td><a href="https://uclahs-cds.github.io/package-BoutrosLab-plotting-general/reference/create.densityplot.html" class="external-link"><code>create.densityplot()</code></a></td>
<td>Make a density plot</td>
</tr>
<tr class="odd">
<td><a href="https://uclahs-cds.github.io/package-BoutrosLab-plotting-general/reference/create.dotmap.html" class="external-link"><code>create.dotmap()</code></a></td>
<td>Make a dotmap with coloured background</td>
</tr>
<tr class="even">
<td><a href="https://uclahs-cds.github.io/package-BoutrosLab-plotting-general/reference/create.gif.html" class="external-link"><code>create.gif()</code></a></td>
<td>Make a gif</td>
</tr>
<tr class="odd">
<td><a href="https://uclahs-cds.github.io/package-BoutrosLab-plotting-general/reference/create.heatmap.html" class="external-link"><code>create.heatmap()</code></a></td>
<td>Make a heatmap</td>
</tr>
<tr class="even">
<td><a href="https://uclahs-cds.github.io/package-BoutrosLab-plotting-general/reference/create.hexbinplot.html" class="external-link"><code>create.hexbinplot()</code></a></td>
<td>Make a hexagonally binned plot</td>
</tr>
<tr class="odd">
<td><a href="https://uclahs-cds.github.io/package-BoutrosLab-plotting-general/reference/create.histogram.html" class="external-link"><code>create.histogram()</code></a></td>
<td>Make a histogram</td>
</tr>
<tr class="even">
<td><a href="https://uclahs-cds.github.io/package-BoutrosLab-plotting-general/reference/create.lollipopplot.html" class="external-link"><code>create.lollipopplot()</code></a></td>
<td>Make a lollipopplot</td>
</tr>
<tr class="odd">
<td><a href="https://uclahs-cds.github.io/package-BoutrosLab-plotting-general/reference/create.manhattanplot.html" class="external-link"><code>create.manhattanplot()</code></a></td>
<td>Make a Manhattan plot</td>
</tr>
<tr class="even">
<td><a href="https://uclahs-cds.github.io/package-BoutrosLab-plotting-general/reference/create.multipanelplot.html" class="external-link"><code>create.multipanelplot()</code></a></td>
<td>Joins plots together</td>
</tr>
<tr class="odd">
<td><a href="https://uclahs-cds.github.io/package-BoutrosLab-plotting-general/reference/create.multiplot.html" class="external-link"><code>create.multiplot()</code></a></td>
<td>Joins plots together</td>
</tr>
<tr class="even">
<td><a href="https://uclahs-cds.github.io/package-BoutrosLab-plotting-general/reference/create.polygonplot.html" class="external-link"><code>create.polygonplot()</code></a></td>
<td>Make a polygonplot</td>
</tr>
<tr class="odd">
<td><a href="https://uclahs-cds.github.io/package-BoutrosLab-plotting-general/reference/create.qqplot.comparison.html" class="external-link"><code>create.qqplot.comparison()</code></a></td>
<td>Make a quantile-quantile plot of two samples</td>
</tr>
<tr class="even">
<td><a href="https://uclahs-cds.github.io/package-BoutrosLab-plotting-general/reference/create.qqplot.fit.html" class="external-link"><code>create.qqplot.fit()</code></a></td>
<td>Make a quantile-quantile plot of a sample</td>
</tr>
<tr class="odd">
<td><a href="https://uclahs-cds.github.io/package-BoutrosLab-plotting-general/reference/create.qqplot.fit.confidence.interval.html" class="external-link"><code>create.qqplot.fit.confidence.interval()</code></a></td>
<td>Create the confidence bands for a one-sample qq plot</td>
</tr>
<tr class="even">
<td><a href="https://uclahs-cds.github.io/package-BoutrosLab-plotting-general/reference/create.scatterplot.html" class="external-link"><code>create.scatterplot()</code></a></td>
<td>Make a scatterplot</td>
</tr>
<tr class="odd">
<td><a href="https://uclahs-cds.github.io/package-BoutrosLab-plotting-general/reference/create.segplot.html" class="external-link"><code>create.segplot()</code></a></td>
<td>Make a segplot</td>
</tr>
<tr class="even">
<td><a href="https://uclahs-cds.github.io/package-BoutrosLab-plotting-general/reference/create.stripplot.html" class="external-link"><code>create.stripplot()</code></a></td>
<td>Make a strip-plot</td>
</tr>
<tr class="odd">
<td><a href="https://uclahs-cds.github.io/package-BoutrosLab-plotting-general/reference/create.violinplot.html" class="external-link"><code>create.violinplot()</code></a></td>
<td>Make a violin plot</td>
</tr>
<tr class="even">
<td><a href="https://uclahs-cds.github.io/package-BoutrosLab-plotting-general/reference/critical.value.ks.test.html" class="external-link"><code>critical.value.ks.test()</code></a></td>
<td>Critical Value for Kolmogorov-Smirnov Test</td>
</tr>
<tr class="odd">
<td><a href="https://uclahs-cds.github.io/package-BoutrosLab-plotting-general/reference/default.colours.html" class="external-link"><code>default.colours()</code></a></td>
<td>Provides default colour schemes.</td>
</tr>
<tr class="even">
<td><a href="https://uclahs-cds.github.io/package-BoutrosLab-plotting-general/reference/display.colours.html" class="external-link"><code>display.colours()</code></a></td>
<td>Function to display R colors, as well as corresponding R grey colours.</td>
</tr>
<tr class="odd">
<td><a href="https://uclahs-cds.github.io/package-BoutrosLab-plotting-general/reference/display.statistical.result.html" class="external-link"><code>display.statistical.result()</code></a></td>
<td>Utility function to display statistical result in a plot</td>
</tr>
<tr class="even">
<td><a href="https://uclahs-cds.github.io/package-BoutrosLab-plotting-general/reference/dist.html" class="external-link"><code>dist()</code></a></td>
<td>Distance Matrix Computation</td>
</tr>
<tr class="odd">
<td><a href="https://uclahs-cds.github.io/package-BoutrosLab-plotting-general/reference/force.colour.scheme.html" class="external-link"><code>force.colour.scheme()</code></a></td>
<td>Based on predefined colour schemes, returns a vector of corresponding colours.</td>
</tr>
<tr class="even">
<td><a href="https://uclahs-cds.github.io/package-BoutrosLab-plotting-general/reference/generate.at.final.html" class="external-link"><code>generate.at.final()</code></a></td>
<td>Generates alternative default tick mark locations for create.densityplot() and create.</td>
</tr>
<tr class="odd">
<td><a href="https://uclahs-cds.github.io/package-BoutrosLab-plotting-general/reference/get.corr.key.html" class="external-link"><code>get.corr.key()</code></a></td>
<td>Correlation Key</td>
</tr>
<tr class="even">
<td><a href="https://uclahs-cds.github.io/package-BoutrosLab-plotting-general/reference/get.correlation.p.and.corr.html" class="external-link"><code>get.correlation.p.and.corr()</code></a></td>
<td>Calculate a correlation and its statistical significance</td>
</tr>
<tr class="odd">
<td><a href="https://uclahs-cds.github.io/package-BoutrosLab-plotting-general/reference/get.defaults.html" class="external-link"><code>get.defaults()</code></a></td>
<td>Get operating system specific default properties</td>
</tr>
<tr class="even">
<td><a href="https://uclahs-cds.github.io/package-BoutrosLab-plotting-general/reference/get.line.breaks.html" class="external-link"><code>get.line.breaks()</code></a></td>
<td>Get line breaks</td>
</tr>
<tr class="odd">
<td><a href="https://uclahs-cds.github.io/package-BoutrosLab-plotting-general/reference/legend.grob.html" class="external-link"><code>legend.grob()</code></a></td>
<td>Generate a legend grob</td>
</tr>
<tr class="even">
<td><a href="https://uclahs-cds.github.io/package-BoutrosLab-plotting-general/reference/panel.BL.bwplot.html" class="external-link"><code>panel.BL.bwplot</code></a></td>
<td>A lattice::panel.bwplot replacement that fixes colouring issues</td>
</tr>
<tr class="odd">
<td><a href="https://uclahs-cds.github.io/package-BoutrosLab-plotting-general/reference/pcawg.colours.html" class="external-link"><code>pcawg.colours()</code></a></td>
<td>Return standard PCAWG colour palettes.</td>
</tr>
<tr class="even">
<td><a href="https://uclahs-cds.github.io/package-BoutrosLab-plotting-general/reference/scientific.notation.html" class="external-link"><code>scientific.notation()</code></a></td>
<td>Use scientific notation in plots</td>
</tr>
<tr class="odd">
<td><a href="https://uclahs-cds.github.io/package-BoutrosLab-plotting-general/reference/show.available.palettes.html" class="external-link"><code>show.available.palettes()</code></a></td>
<td>Display the available colour palettes</td>
</tr>
<tr class="even">
<td><a href="https://uclahs-cds.github.io/package-BoutrosLab-plotting-general/reference/thousands.split.html" class="external-link"><code>thousands.split()</code></a></td>
<td>Divide strings into groups of thousands</td>
</tr>
<tr class="odd">
<td><a href="https://uclahs-cds.github.io/package-BoutrosLab-plotting-general/reference/write.metadata.html" class="external-link"><code>write.metadata()</code></a></td>
<td>Writes Metadata</td>
</tr>
<tr class="even">
<td><a href="https://uclahs-cds.github.io/package-BoutrosLab-plotting-general/reference/write.plot.html" class="external-link"><code>write.plot()</code></a></td>
<td>Simplifies plotting by standardizing and centralizing all output-handling</td>
</tr>
</tbody>
</table>
</div>
<div class="section level2">
<h2 id="included-datasets">Included datasets<a class="anchor" aria-label="anchor" href="#included-datasets"></a>
</h2>
<table class="table">
<colgroup>
<col width="55%">
<col width="45%">
</colgroup>
<thead><tr class="header">
<th><strong>Dataset</strong></th>
<th><strong>Description</strong></th>
</tr></thead>
<tbody>
<tr class="odd">
<td><a href="https://uclahs-cds.github.io/package-BoutrosLab-plotting-general/reference/CNA.html" class="external-link"><code>CNA</code></a></td>
<td>Copy number aberration (CNA) data from colon cancer patients</td>
</tr>
<tr class="even">
<td><a href="https://uclahs-cds.github.io/package-BoutrosLab-plotting-general/reference/SNV.html" class="external-link"><code>SNV</code></a></td>
<td>Single nucleotide variant (SNV) data from colon cancer patients</td>
</tr>
<tr class="odd">
<td><a href="https://uclahs-cds.github.io/package-BoutrosLab-plotting-general/reference/microarray.html" class="external-link"><code>microarray</code></a></td>
<td>Microarray dataset of colon cancer patients</td>
</tr>
<tr class="even">
<td><a href="https://uclahs-cds.github.io/package-BoutrosLab-plotting-general/reference/patient.html" class="external-link"><code>patient</code></a></td>
<td>Dataset describing qualities of 58 colon cancer patients</td>
</tr>
</tbody>
</table>
</div>
<div class="section level2">
<h2 id="resources">Resources<a class="anchor" aria-label="anchor" href="#resources"></a>
</h2>
<p>Available resources on BPG usage can be found at the package <a href="https://cran.r-project.org/package=BoutrosLab.plotting.general/index.html" class="external-link">CRAN page</a>, <a href="https://cran.r-project.org/package=BoutrosLab.plotting.general/BoutrosLab.plotting.general.pdf" class="external-link">reference manual</a>, or <a href="https://cran.r-project.org/package=BoutrosLab.plotting.general/vignettes/PlottingGuide.pdf" class="external-link">vignette</a>.</p>
</div>
<div class="section level2">
<h2 id="getting-help">Getting help<a class="anchor" aria-label="anchor" href="#getting-help"></a>
</h2>
<p>Looking for guidance or support with Boutros.plotting.general? Check out our <a href="https://github.com/uclahs-cds/package-BoutrosLab-plotting-general/discussions" class="external-link">Discussions</a> page.</p>
<p>Submit bugs, suggest new features or see current works at our <a href="https://github.com/uclahs-cds/package-BoutrosLab-plotting-general/issues" class="external-link">Issues</a> page.</p>
<p><a href="https://github.com/uclahs-cds/package-BoutrosLab-plotting-general/pulls" class="external-link">Pull requests</a> are also open for discussion.</p>
</div>
<div class="section level2">
<h2 id="contributors">Contributors<a class="anchor" aria-label="anchor" href="#contributors"></a>
</h2>
<p>For lists of contributors please visit <a href="https://uclahs-cds.github.io/package-BoutrosLab-plotting-general/authors.html" class="external-link">here</a> and <a href="https://github.com/uclahs-cds/package-BoutrosLab-plotting-general/graphs/contributors" class="external-link">here</a> at GitHub</p>
</div>
<div class="section level2">
<h2 id="citation-information">Citation information<a class="anchor" aria-label="anchor" href="#citation-information"></a>
</h2>
<p>To cite package <code>BoutrosLab.plotting.general</code> in publications use:</p>
<p>P’ng, C., Green, J., Chong, L.C. et al. <em>BPG: Seamless, automated and interactive visualization of scientific data</em>. BMC Bioinformatics 20, 42 (2019). <a href="https://doi.org/10.1186/s12859-019-2610-2" class="external-link uri">https://doi.org/10.1186/s12859-019-2610-2</a></p>
<p>A BibTeX entry for LaTeX users is</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode bibtex"><code class="sourceCode bibtex"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="co">  </span><span class="va">@Article</span>{,</span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a>    <span class="dt">title</span> = {BPG: Seamless, automated and interactive visualization of scientific data},</span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a>    <span class="dt">journal</span> = {BMC Bioinformatics},</span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a>    <span class="dt">doi</span> = {10.1186/s12859-019-2610-2},</span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a>    <span class="dt">url</span> = {https://doi.org/10.1186/s12859-019-2610-2},</span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a>    <span class="dt">volume</span> = {20},</span>
<span id="cb3-7"><a href="#cb3-7" tabindex="-1"></a>    <span class="dt">number</span> = {42},</span>
<span id="cb3-8"><a href="#cb3-8" tabindex="-1"></a>    <span class="dt">year</span> = {2019},</span>
<span id="cb3-9"><a href="#cb3-9" tabindex="-1"></a>    <span class="dt">month</span> = {January},</span>
<span id="cb3-10"><a href="#cb3-10" tabindex="-1"></a>    <span class="dt">day</span> = {21},</span>
<span id="cb3-11"><a href="#cb3-11" tabindex="-1"></a>    <span class="dt">issn</span> = {1471-2105},</span>
<span id="cb3-12"><a href="#cb3-12" tabindex="-1"></a>    <span class="dt">author</span> = {Christine P'ng and Jeffrey Green and Lauren C. Chong and Daryl Waggott and Stephenie D. Prokopec and Mehrdad Shamsi and Francis Nguyen and Denise Y. F. Mak and Felix Lam and Marco A. Albuquerque and Ying Wu and Esther H. Jung and Maud H. W. Starmans and Michelle A. Chan-Seng-Yue and Cindy Q. Yao and Bianca Liang and Emilie Lalonde and Syed Haider and Nicole A. Simone and Dorota Sendorek and Kenneth C. Chu and Nathalie C. Moon and Natalie S. Fox and Michal R. Grzadkowski and Nicholas J. Harding and Clement Fung and Amanda R. Murdoch and Kathleen E. Houlahan and Jianxin Wang and David R. Garcia and Richard de Borja and Ren X. Sun and Xihui Lin and Gregory M. Chen and Aileen Lu and Yu-Jia Shiah and Amin Zia and Ryan Kearns and Paul C. Boutros}</span>
<span id="cb3-13"><a href="#cb3-13" tabindex="-1"></a>  }</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="license">License<a class="anchor" aria-label="anchor" href="#license"></a>
</h2>
<p>Authors: Christine P’ng, Jeffrey Green, Lauren C. Chong, Daryl Waggott, Stephenie D. Prokopec, Mehrdad Shamsi, Francis Nguyen, Denise Y. F. Mak, Felix Lam, Marco A. Albuquerque, Ying Wu, Esther H. Jung, Maud H. W. Starmans, Michelle A. Chan-Seng-Yue, Cindy Q. Yao, Bianca Liang, Emilie Lalonde, Syed Haider, Nicole A. Simone, Dorota Sendorek, Kenneth C. Chu, Nathalie C. Moon, Natalie S. Fox, Michal R. Grzadkowski, Nicholas J. Harding, Clement Fung, Amanda R. Murdoch, Kathleen E. Houlahan, Jianxin Wang, David R. Garcia, Richard de Borja, Ren X. Sun, Xihui Lin, Gregory M. Chen, Aileen Lu, Yu-Jia Shiah, Amin Zia, Ryan Kearns, Stefan Eng, Dan Knight, Julie Livingstone, Chase Bakkeby, Rupert Hugh-White, Adriana Salcedo, Takafumi Yamaguchi, Mohammed Faizal Eeman Mootor, Chenghao Zhu, Caden Bugh &amp; Paul C. Boutros (<a href="mailto:PBoutros@mednet.ucla.edu" class="email">PBoutros@mednet.ucla.edu</a>)</p>
<p>BoutrosLab.plotting.general is licensed under the GNU General Public License version 2. See the file LICENSE.md for the terms of the GNU GPL license.</p>
<p>BoutrosLab.plotting.general is a software package for generating publication-quality, customizable plots.</p>
<p>Copyright (C) 2014-2018 Ontario Institute for Cancer Research and 2018-2023 University of California Los Angeles (“Boutros Lab”) All rights reserved.</p>
<p>This program is free software; you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation; either version 2 of the License, or (at your option) any later version.</p>
<p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.</p>
</div>
</div>
  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://cloud.r-project.org/package=BoutrosLab.plotting.general" class="external-link">View on CRAN</a></li>
<li><a href="https://github.com/uclahs-cds/package-BoutrosLab-plotting-general/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/uclahs-cds/package-BoutrosLab-plotting-general/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="https://www.r-project.org/Licenses/GPL-2" class="external-link">GPL-2</a></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing BoutrosLab.plotting.general</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Paul Boutros <br><small class="roles"> Author, maintainer </small>   </li>
<li><a href="authors.html">More about authors...</a></li>
</ul>
</div>



  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Paul Boutros.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
